src/magic_folder/web/magic_folder.py

```diff
@@ -72,42 +77,41 @@ def __init__(self, get_magic_folder, get_auth_token):
         self.get_auth_token = get_auth_token
 
     def render_POST(self, request):
-        authorize(request, self.get_auth_token)
+        if not authorize(request, self.get_auth_token):
+            return NOT_DONE_YET
 
         request.setHeader("content-type", "application/json")
         nick = request.args.get("name", ["default"])[0]
 
         try:
             magic_folder = self.get_magic_folder(nick)
         except KeyError:
-            raise WebError(
-                "No such magic-folder '{}'".format(nick),
-                404,
-            )
+            request.setResponseCode(http.NOT_FOUND)
+            return json.dumps({
+                u"error":
+                u"No such magic-folder: {}".format(
+                    nick.decode("utf-8"),
+                ),
+            })
 
         data = []
         for item in magic_folder.uploader.get_status():
-            d = dict(
-                path=item.relpath_u,
-                status=item.status_history()[-1][0],
-                kind='upload',
-            )
-            for (status, ts) in item.status_history():
-                d[status + '_at'] = ts
-            d['percent_done'] = item.progress.progress
-            d['size'] = item.size
-            data.append(d)
+            data.append(status_for_item("upload", item))
 
         for item in magic_folder.downloader.get_status():
-            d = dict(
-                path=item.relpath_u,
-                status=item.status_history()[-1][0],
-                kind='download',
-            )
-            for (status, ts) in item.status_history():
-                d[status + '_at'] = ts
-            d['percent_done'] = item.progress.progress
-            d['size'] = item.size
-            data.append(d)
+            data.append(status_for_item("download", item))
 
         return json.dumps(data)
+
+
+def status_for_item(kind, item):
+    d = dict(
+        path=item.relpath_u,
+        status=item.status_history()[-1][0],
+        kind='upload',
```

Comment:
Don't we need to assign `kind` to the passed parameter `kind` instead of `upload`?

Reply:
Oops!  Yes, thanks.  Nice catch.